File: src/clang/magma/compile/lang/SymbolFilter.c
#include "SymbolFilter.h"
int isValid(String input){if(input.isEmpty()) return false;for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);
            if (c == '_' || Character.isLetter(c) || (i != 0 && Character.isDigit(c))) continue;
            return false;
        }return true;
}

File: src/java/magmac/app/compile/rule/SymbolFilter.java
package magmac.app.compile.rule;

public class SymbolFilter implements Filter {
    @Override
    public boolean test(String input) {
        int length = input.length();
        for (int i = 0; i < length; i++) {
            char c = input.charAt(i);
            if (Character.isLetter(c) || (0 != i && Character.isDigit(c))) {
                continue;
            }
            return false;
        }
        return true;
    }

    @Override
    public String createMessage() {
        return "Not a symbol";
    }
}